packages/platforms/dni/x86-64/x86-64-dni-wb2448/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,173 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *        Copyright 2014, 2015 Big Switch Networks, Inc.
+ *        Copyright 2017 Delta Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ *
+ ***********************************************************/
+#include <onlp/platformi/sysi.h>
+#include <onlplib/crc32.h>
+#include <onlplib/file.h>
+
+#include "x86_64_dni_wb2448_log.h"
+#include "platform_lib.h"
+
+#define IDPROM_PATH 					"/sys/class/i2c-adapter/i2c-0/0-0053/eeprom"
+
+#define NUM_OF_THERMAL_ON_MAIN_BROAD	5
+#define NUM_OF_LED_ON_MAIN_BROAD		5
+#define NUM_OF_PSU_ON_MAIN_BROAD		0
+#define NUM_OF_FAN_ON_MAIN_BROAD		2
+
+#define I2C_BUS					0x04
+#define CPLD_ADDR				0x28
+#define CPLD_VERSION_REGISTER	0x08
+#define CPLD_VERSION_ADDR_LEN	0x01
+#define CPLD_VERSION_DATA_LEN	0x01
+#define CPLD_VERSION_OFFSET		5
+
+const char*
+onlp_sysi_platform_get(void)
+{
+    return "x86-64-dni-wb2448-r0";
+}
+
+int
+onlp_sysi_init(void)
+{
+    return ONLP_STATUS_OK;
+}
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{
+    uint8_t* rdata = aim_zmalloc(256);
+	
+    if(onlp_file_read(rdata, 256, size, IDPROM_PATH) == ONLP_STATUS_OK) 
+	{
+		if(*size == 256) 
+		{
+            *data = rdata;
+            return ONLP_STATUS_OK;
+        }
+    }
+    
+	aim_free(rdata);
+    *size = 0;
+	
+    return ONLP_STATUS_E_INTERNAL;
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{
+    INT4 rv		 = ONLP_STATUS_OK;
+	UINT4 u4Data = 0;
+    
+	rv = ifnOS_LINUX_BmcI2CGet(I2C_BUS, CPLD_ADDR, CPLD_VERSION_REGISTER, CPLD_VERSION_ADDR_LEN, &u4Data, CPLD_VERSION_DATA_LEN);
+	
+	if(rv == ONLP_STATUS_OK)
+	{
+		u4Data = u4Data >> CPLD_VERSION_OFFSET;
+		
+		pi->cpld_versions = aim_fstrdup("%d", u4Data);
+    }
+	    
+	return rv;
+}
+
+void
+onlp_sysi_platform_info_free(onlp_platform_info_t* pi)
+{
+    aim_free(pi->cpld_versions);
+}
+
+void
+onlp_sysi_onie_data_free(uint8_t* data)
+{
+    /*
+     * We returned a static array in onlp_sysi_onie_data_get()
+     * so no free operation is required.
+     */
+}
+
+int 
+onlp_sysi_onie_info_get(onlp_onie_info_t* onie)
```

Comment:
@vincent0083  This function is not needed if you provide a proper implementation for ``onlp_sysi_onie_data_get``. I suggest you remove it to avoid confusion.